astroquery/mast/observations.py

```diff
@@ -440,6 +440,8 @@ def get_product_list_async(self, observations):
             observations = np.array([observations])
         if isinstance(observations, Table):
             observations = observations['obsid']
+        if isinstance(observations, list):
+            observations = np.array(observations)
```

Comment:
This looks good in that it will address the case where `observations` came in as a list.  I did notice a weird edge case however.

If observations comes in as an empty list, it gets converted to `array([], dtype=float64)`.  For that empty NDArray (and presumable empty NDArrays that weren't converted from lists), the next line has weird behavior:
observations = observations[observations != ""]
It issues a future warning, and returns an empty NDArray where `len(observations)==1`.  This makes the `len()==0` check fail, so there's a messy error later.

I'm not an expert on numpy array filtering, but if we're assuming that `observations` is a numpy array at this point, maybe we should do an explicit check for that after the if block, then maybe use `observations.size==0` instead of `len()==0`?

Reply:
For the `observations.size==0`, the filtering could just be skipped I guess.

Reply:
Thanks Tom! I agree - better to air on the safe side. I've added your two suggestions.

Reply:
Thanks, @jaymedina !  In looking again, I think I oversimplified it a little.  I think there are other types of incoming `observations` that need to be covered (like Astropy Table or Row).  I would generally suggest using `instanceof` instead of `type() == ` to make sure inherited types are covered.   

I had been thinking that Astropy tables inherited from `np.ndarray`, but they don't, so if there's type checking here it needs to be a little messier, and worse, my suggestion of `observations.size` becomes bogus.  Sorry about that!!  I guess the bottom line is that since this method can accept multiple types for `observations`, we need to make sure that all those possibilities are handled cleanly.  The unit tests are probably a good place to ensure that handling.

Reply:
Hi Tom, no worries, and yep I just realized that as well when trying to run my unit test and got an error for the Astropy Table and Row cases, so I decided to remove the assertion. I agree we can rely on the unit tests for these kinds of checks, so I added a new one for the input list case, and I think that should give us ample coverage. Thanks for your input!